ui/raidboss/data/06-ew/ultimate/the_omega_protocol.ts

```diff
@@ -1366,6 +1366,368 @@ const triggerSet: TriggerSet<Data> = {
+          dir1 = pos1 === 92.93 ? output.dirN!() : output.dirS!();
+          dir2 = pos2 === 92.93 ? output.dirW!() : output.dirE!();
+        }
+
+        // Secondary Spot for Staff + Sword
+        if (!isM1In && !isF1In) {
+          if (matches.id === '7B9B') {
+            // East/West Safe
+            if (f1.PosX === 92.93 && f1.PosY === 92.93) {
+              // NW
+              dir3 = output.dirWSW!();
+            } else if (f1.PosX === 92.93 && f1.PosY !== 92.93) {
+              // SW
+              dir3 = output.dirWNW!();
+            } else if (f1.PosX !== 92.93 && f1.PosY === 92.93) {
+              // NE
+              dir3 = output.dirESE!();
+            } else {
+              // SE
+              dir3 = output.dirENE!();
+            }
+          } else {
+            // North/South Safe
+            if (f1.PosX === 92.93 && f1.PosY === 92.93) {
+              // NW
+              dir3 = output.dirNNE!();
+            } else if (f1.PosX === 92.93 && f1.PosY !== 92.93) {
+              // SW
+              dir3 = output.dirSSE!();
+            } else if (f1.PosX !== 92.93 && f1.PosY === 92.93) {
+              // NE
+              dir3 = output.dirNNW!();
+            } else {
+              // SE
+              dir3 = output.dirSSW!();
+            }
+          }
+        }
+        if (!isM2In && !isF2In) {
+          if (matches.id === '7B9B') {
+            // East/West Safe
+            if (f2.PosX === 92.93 && f2.PosY === 92.93) {
+              // NW
+              dir4 = output.dirNNE!();
+            } else if (f2.PosX === 92.93 && f2.PosY !== 92.93) {
+              // SW
+              dir4 = output.dirSSE!();
+            } else if (f2.PosX !== 92.93 && f2.PosY === 92.93) {
+              // NE
+              dir4 = output.dirNNW!();
+            } else {
+              // SE
+              dir4 = output.dirSSW!();
+            }
+          } else {
+            // North/South Safe
+            if (f2.PosX === 92.93 && f2.PosY === 92.93) {
+              // NW
+              dir4 = output.dirWSW!();
+            } else if (f2.PosX === 92.93 && f2.PosY !== 92.93) {
+              // SW
+              dir4 = output.dirWNW!();
+            } else if (f2.PosX !== 92.93 && f2.PosY === 92.93) {
+              // NE
+              dir4 = output.dirESE!();
+            } else {
+              // SE
+              dir4 = output.dirENE!();
+            }
+          }
+        }
+
+        const firstSpot = output.safeSpot!({ distance: distance1, dir: dir1 });
+        const secondSpot = output.safeSpot!({ distance: distance2, dir: dir2 });
+
+        if (!isM1In && !isF1In && !isM2In && !isF2In) {
+          // Output two locations => two locations
+          const staffSwordFar1 = output.staffSwordFar!({ dir: dir1 });
+          const staffSwordFar2 = output.staffSwordFar!({ dir: dir2 });
+          const staffSwordMid1 = output.staffSwordMid!({ dir: dir3 });
```

Comment:
My only feedback here is if you can use a different output string for this "mid".  I think it's a different "mid" then the other one and I (in particular) want to use a different word there to differentiate.  You could call it `midStaff` or something?